### PR TITLE
Add confirmation email

### LINKED ISSUE #53: Requirements checklist for MVP
- [x] Rails update to latest version 6.1.3
- [x] User confirmation after registration
  - We could do this either through a confirmation e-mail or manually in a first phase
- [x] Add authorization policies for all eligible actions
- [ ] Add error monitoring tool for production environments 
  - [Honeybadger](https://www.honeybadger.io/plans/) allows up to 3 users for free
- [x] Any user, in any of their teams (later, we'll enforce a fairness rule, but for now let's do it like this), can pick THE movie for that round. There's only one, we may or may not allow to replace; up to debate
- [x] Fix round number in views (currently using id)
- [x] Add `admin` flag to the user model (this flag can be used to give super-user permissions, like editing protected data, etc.)
- [ ] Improve homepage's "copy"
- [ ] [to evaluate further] asynchronous/background job processing (namely for sending confirmation e-mail)
- [ ] Add privacy policy page
- [ ] Configure logging. Study options. By this I mean writing creating meaningful messages for several events, like HTTP requests made to the outside, execution stages of interactors, etc. The logger should write relevant tags for each line like HTTP request UUID, timestamp, etc. This is to be later integrated with Papertrail or similar.
- [ ] Control rounds' membership: users that belong to the round (initially the user which creates it, its only member) can invite other users, only then can they join. (I think in this phase we don't need to think about "team administrators" different than other members. This goes along the model of users are well-behaved **inside their teams** (so we only need to protect other teams).

### PR DESCRIPTION
**Pull request by gdiasdasilva**
This pull request adds:
* A template/view for the confirmation e-mail, including a link to confirm the e-mail address;
* A new page that render the token in an input field by fetching the token from a `token` query param;
* An endpoint to which the previously mentioned token may be submitted for validation and potential confirmation of the user
* Multiple specs

We should also create an end-to-end feature test with the following flow:

1. User signs up
2. User checks e-mail and clicks the link
3. User submits the token in the token confirmation page
4. User is confirmed

I'd say this can be done in a separate pull request, as it will require some extra investigation.

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -4,7 +4,8 @@ def create
     @user.email_confirmation_token = Clearance::Token.new
 
     if @user.save
-      redirect_back_or url_after_create
+      UserMailer.confirmation_email(@user).deliver_now # should be done in a background job when possible
```
Comment (ruiccarvalho) [inline app/controllers/users_controller.rb]: Please, update the comment to be a proper `TODO: `

**Review by ruiccarvalho — CHANGES_REQUESTED** [review]
One very tiny note and, for now, just a remark: I think it's a good practice to add a plaintext version of the templates, to be sent together with the HTML.

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (gdiasdasilva) [reply app/controllers/users_controller.rb]: Done

**Review by ruiccarvalho — APPROVED** [review]
LGTM

**Comment by ruiccarvalho** [comment]
As agreed, performing the merging myself, since there are no requested changes.

**Pull request merged**